s3/s3_transport/include/s3_transport.hpp

```diff
@@ -1868,23 +1953,25 @@ namespace irods::experimental::io::s3_transport
                 determine_start_and_end_part_from_offset_and_bytes_this_thread(bytes_this_thread, file_offset_,
                         config_.circular_buffer_size, start_part_number, end_part_number, part_sizes);
 
-                // estimate the size and resize the etags vector
-                unsigned int total_number_of_parts = config_.object_size / bytes_this_thread;
-                total_number_of_parts = total_number_of_parts < end_part_number ? end_part_number : total_number_of_parts;
-
                 // resize the etags vector if necessary
-                int resize_error = shm_obj.atomic_exec([this, total_number_of_parts, &shm_obj](auto& data) {
+                int resize_error = shm_obj.atomic_exec([this, &shm_obj](auto& data) {
+
+                    if (constants::MAXIMUM_NUMBER_ETAGS_PER_UPLOAD > data.etags.size()) {
+
+                        rodsLog(config_.developer_messages_log_level,  "%s:%d (%s) [[%lu]] resize etags vector from %lu to %u\n",
+                                __FILE__, __LINE__, __FUNCTION__, get_thread_identifier(), data.etags.size(), constants::MAXIMUM_NUMBER_ETAGS_PER_UPLOAD);
 
-                    if (total_number_of_parts > data.etags.size()) {
                         try {
-                            data.etags.resize(total_number_of_parts, types::shm_char_string("", shm_obj.get_allocator()));
-                        } catch (std::bad_alloc& ba) {
+                            data.etags.resize(constants::MAXIMUM_NUMBER_ETAGS_PER_UPLOAD, types::shm_char_string("", shm_obj.get_allocator()));
+                        } catch (boost::interprocess::bad_alloc &biba) {
                             this->set_error(ERROR(S3_PUT_ERROR, "Error on reallocation of etags buffer in shared memory."));
                             data.last_error_code = error_codes::BAD_ALLOC;
                             return true;
                         }
+
                     }
 
+
                     return false;
 
```

Comment:
Would it make sense to move this lambda into a method since it's used in multiple places?

Reply:
There should have only been one as I moved it up to run on both cache and cacheless flows.  Removing the second one as it is unnecessary.